src/main/scala/kamon/statsd/SimpleMetricKeyGenerator.scala

```diff
@@ -44,8 +44,8 @@ class SimpleMetricKeyGenerator(config: Config) extends MetricKeyGenerator {
     else application
 
   def generateKey(name: String, tags: Map[String, String]): String = {
-    val stringTags = TreeMap(tags.toSeq:_ *).values.map(normalizer).mkString(".")
-    s"$baseName.${normalizer(name)}.$stringTags"
+    val stringTags = if (tags.nonEmpty) "." + TreeMap(tags.toSeq:_ *).values.map(normalizer).mkString(".") else ""
```

Comment:
I'm not sure if this strategy for turning tags into a string is the most appropriate. If we assume that all time series for a given metric are going to have the same set of tags, just different values then this sort of could work, but its missing the part of knowing what does each value represent. What about encoding this as "tagName.tagValue" and repeat till the end of tags? Also, tags should be ordered before this to ensure they will always be matching positions with all other time series of the same metric.

Reply:
I didn't write this logic and I have found it a bit odd. :) I like idea and I'm going to implement it but it will not be the implementation I will use because in the `Telegraf` world the tag are comma separated values with the following format.
```
matricName,tagName1=tagValue1,tagName2=tagValue2
```